tests/functional/test_completion.py

```diff
@@ -113,14 +114,97 @@ def test_completion_for_default_parameters(script):
 
 def test_completion_option_for_command(script):
     """
-    Test getting completion for ``--`` in command (eg. pip search --)
+    Test getting completion for ``--`` in command (e.g. ``pip search --``)
     """
 
     res, env = setup_completion(script, 'pip search --', '2')
     assert '--help' in res.stdout,\
            "autocomplete function could not complete ``--``"
 
 
+def test_completion_files_after_option(script, data):
+    """
+    Test getting completion for <file> or <dir> after options in command
+    (e.g. ``pip install -r``)
+    """
+    res, env = setup_completion(
+        script, ('pip install -r r'),
+        '3',
+        data.complete_paths
+    )
+    assert 'requirements.txt' in res.stdout,\
+        "autocomplete function could not complete <file>"\
+        "after options in command"
+    assert 'README.txt' not in res.stdout,\
+        "autocomplete function completed <file> that"\
+        "should not be completed"
+    assert 'resources' in res.stdout,\
+        "autocomplete function could not complete <dir>"\
+        "after options in command"
+
+
+def test_completion_not_files_after_option(script, data):
+    """
+    Test not getting completion files after options which not applicable
+    (e.g. ``pip install``)
+    """
+    res, env = setup_completion(
+        script, ('pip install r'),
+        '2',
+        data.complete_paths
+    )
+    assert 'requirements.txt' not in res.stdout,\
```

Comment:
Also, assert that `README.txt` doesn't show up, when normalisation of case might allow?